File: back-end/services/auth-identity-service/src/main/java/com/codeshare/airline/auth/service/TenantService.java
package com.codeshare.airline.auth.service;


import com.codeshare.airline.auth.authentication.domain.model.TenantContext;
import com.codeshare.airline.core.dto.tenant.TenantDTO;

import java.util.List;
import java.util.UUID;

public interface TenantService {

    TenantDTO create(TenantDTO dto);

    TenantDTO update(UUID id, TenantDTO dto);

    TenantDTO getById(UUID id);

    TenantDTO getByCode(String code);

    List<TenantDTO> getAll();

    void delete(UUID id);

    TenantContext getByTenantCode(String tenantCode);
}

File: front-end/src/app/features/iam/tenants/services/tenant.service.ts
import {inject, Injectable} from '@angular/core';
import {AppApiService} from '@core/config/app-api.service';
import {AppToastService} from '@core/services/app-toast.service';
import {tap} from 'rxjs';
import {Tenant} from "@features/iam/models/tenant.model";

@Injectable({ providedIn: 'root' })
export class TenantService  {

    private api = inject(AppApiService);
    private toast = inject(AppToastService);

    getAll() {
        return this.api.get<Tenant[]>('tenants.base');
    }

    getById(id: string) {
        return this.api.get<Tenant>('tenants.byId', {
            pathParams: { id }
        });
    }

    create(tenant: Tenant) {
        return this.api.post<Tenant>('tenants.base', tenant).pipe(
            tap(() => {
                this.toast.success('Tenant created successfully');
            })
        );
    }

    update(id: string, tenant: Tenant) {
        return this.api.put<Tenant>('tenants.byId', tenant, {
            pathParams: { id }
        }).pipe(
            tap(() => {
                this.toast.success('Tenant updated successfully');
            })
        );
    }

    delete(id: string) {
        return this.api.delete<void>('tenants.byId', {
            pathParams: { id }
        }).pipe(
            tap(() => {
                this.toast.success('Tenant deleted successfully');
            })
        );
    }
}
